### ISSUE TITLE
Alias should be unique per alert occurrence for Opsgenie

### ISSUE THREAD

**Issue by aspyct** [prometheus/alertmanager#2104]
Opsgenie deduplicates alerts based on their `alias` as provided by alertmanager. Two different occurrences of the same alert will have the same alias, even though they are "reopened issues". Because of that, Opsgenie will not send an alert for the subsequent alerts, even though they are separate problems.

Maybe the alert's "active since" should be included in the alias, so that each occurrence of an alert would have a different alias.

Alternatively, if there was a way to set labels based on expressions (`my_label: time()`), that could also be a good solution. We spent some time looking for it, it doesn't seem to be possible.

**What did you do?**
We integrated alertmanager with Opsgenie. We tested it by triggering an alert, fixing it, and triggering the same alert again, as if the problem reoccurred.

**What did you expect to see?**
Opsgenie should have called us twice, once for each alert sent by alertmanager.

**What did you see instead? Under which circumstances?**
The second alert was filtered out by Opsgenie because the `alias` is the same as the first occurrence of the alert.

**Environment**

* System information:

Running inside docker containers on kubernetes.

* Alertmanager version:

```
alertmanager, version 0.19.0 (branch: HEAD, revision: 7aa5d19fea3f58e3d27dbdeb0f2883037168914a)
  build user:       root@587d0268f963
  build date:       20190903-15:01:40
  go version:       go1.12.8
```

* Alertmanager configuration file:
```
route:
  receiver: opsgenie-p3
  group_by:
  - job
  - alertname
  - env
  ...
```

**Comment by simonpasquier** [comment]
Can you provide the configuration of the `opsgenie-p3` receiver? If you set `send_resolved: true` (which is the default), Alertmanager should send the resolved notification and OpsGenie should close the alert. If the alert fires again, it should create a new alert in OpsGenie.

**Comment by aspyct** [comment]
We could confirm that the resolve notification is sent to Opsgenie.

When the alert fires again, it fires with the same alias as the previous alert. Opsgenie then concludes it is simply an echo of an old alert, and discards it.

```    receivers:
      - name: 'opsgenie-p3'
        opsgenie_configs:
          - priority: P3```

**Comment by brian-brazil** [comment]
This disagrees with the docs: https://docs.opsgenie.com/docs/alert-deduplication

This is only for opened issues, in which case this is a feature request to OpsGenie.

**Comment by simonpasquier** [comment]
I did a quick test and I've received a notification from OpsGenie after the alert group went firing, then resolved and firing again. There must be something off with your OpsGenie settings.

**Comment by aspyct** [comment]
Yes, I just got it working, thanks for your support.

It turns out I had multiple small issues clouding the whole thing:
1. we have 3 environments (dev, acc and prd) all integrating with Opsgenie. It turns out all three envs had the same aliases for the same alerts. I needed labels for each env (see https://github.com/prometheus/alertmanager/issues/1598#issuecomment-473527548)
2. on top of that, I didn't even have a group_by when I tested on Friday, and thus ALL alerts had the same alias (because it's just hashing... nothing)
3. The alerts take a long time to propagate all the way from prometheus to opsgenie. I will now look into that.

Thank you for your guidance with this issue!
